modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/upgrade/v14_0_0/test/ObjectDefinitionUpgradeProcessTest.java

```diff
@@ -0,0 +1,456 @@
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ProxyUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.io.Serializable;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Leite
+ */
+@RunWith(Arquillian.class)
+public class ObjectDefinitionUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_originalSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager(
+					CPDefinition.class.getSimpleName());
+
+		SystemObjectDefinitionManager newSystemObjectDefinitionManager =
+			(SystemObjectDefinitionManager)ProxyUtil.newProxyInstance(
+				SystemObjectDefinitionManager.class.getClassLoader(),
+				new Class<?>[] {SystemObjectDefinitionManager.class},
+				(proxy, method, args) -> {
+					if (Objects.equals(
+							method.getName(), "getPrimaryKeyColumn")) {
+
+						return CPDefinitionTable.INSTANCE.CPDefinitionId;
+					}
+
+					if (Objects.equals(method.getName(), "getObjectFields")) {
+						List<ObjectField> objectFields =
+							(List<ObjectField>)method.invoke(
+								_originalSystemObjectDefinitionManager, args);
+
+						objectFields.set(
+							4,
+							new TextObjectFieldBuilder(
+							).dbColumnName(
+								"CPDefinitionId"
+							).labelMap(
+								LocalizedMapUtil.getLocalizedMap(
+									RandomTestUtil.randomString())
+							).name(
+								"productId"
+							).system(
+								true
+							).build());
+
+						return objectFields;
+					}
+
+					return method.invoke(
+						_originalSystemObjectDefinitionManager, args);
+				});
+
+		_serviceTrackerMap = ReflectionTestUtil.getFieldValue(
+			_systemObjectDefinitionManagerRegistry, "_serviceTrackerMap");
+
+		ReflectionTestUtil.setFieldValue(
+			_systemObjectDefinitionManagerRegistry, "_serviceTrackerMap",
```

Comment:
@pedro-oliveira446 can we use `getAndSetFieldValue` here?

Reply:
Yes, just done,